### PR TITLE
SYSENG-1230: fix missing LBaaS Backend attributes for Create operations

### PR DESCRIPTION
**Pull request by LittleFox94**
### Description

When creating a Backend, only a certain subset of attributes was sent to the Engine.

Also improves the request body handling for all the LBaaS objects, unifying Create and Update and omitting the State attribute instead of choosing the correct one to send and adds some `omitempty` tags to Object fields.

### Checklist

* [x] added release notes to `Unreleased` section in [CHANGELOG.md](CHANGELOG.md), if user facing change

### References
* [SYSENG-1230](https://ats.anexia-it.com/browse/SYSENG-1230)

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

**Comment by LittleFox94** [comment]
I'll merge this myself after approval, maybe I can prevent swamping the queue

File: pkg/apis/lbaas/v1/helper.go
Comment (marioreggiori) [inline pkg/apis/lbaas/v1/helper.go:18]: return `nil` on error

**Review by marioreggiori — CHANGES_REQUESTED** [review]

File: pkg/apis/lbaas/v1/helper.go
Comment (LittleFox94) [reply pkg/apis/lbaas/v1/helper.go]: fixed :)

**Review by marioreggiori — APPROVED** [review]
lgtm

**Comment by toothstone** [comment]
@LittleFox94 Could you have a quick look if this is also already doing https://ats.anexia-it.com/browse/SYSENG-1013 ? AFAIK LBaaS is the only Generic Service wrapped in go-anxcloud so far, and I'm not aware of any other "optional" fields that had to be sent in earlier Engine versions.

**Comment by LittleFox94** [comment]
> @LittleFox94 Could you have a quick look if this is also already doing https://ats.anexia-it.com/browse/SYSENG-1013 ? AFAIK LBaaS is the only Generic Service wrapped in go-anxcloud so far, and I'm not aware of any other "optional" fields that had to be sent in earlier Engine versions.

one of the functional changes of this PR is to not send the `state` attribute in request bodies anymore, which probably matches SYSENG-1013. We also do not send some empty fields anymore, but that's probably another thing.

File: pkg/apis/lbaas/v1/loadbalancer_types.go
Comment (marioreggiori) [inline pkg/apis/lbaas/v1/loadbalancer_types.go:5]: Are these hooks correkt? I can only find `FilterAPIResponse` on `commonMethods`.

Comment (marioreggiori) [reply pkg/apis/lbaas/v1/loadbalancer_types.go:5]: NVM.. found them

**Comment by marioreggiori** [comment]
lgtm

**Pull request merged**
